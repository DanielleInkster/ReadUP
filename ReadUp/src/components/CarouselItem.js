import React from 'react';
import {
  View,
  Text,
  Image,
  TouchableOpacity,
  Linking,
  StyleSheet,
  ImageBackground,
} from 'react-native';

function trim(item, limit) {
  return item.length > limit ? item.substring(0, limit).trim() + '...' : item;
}

const CarouselItem = ({item, index}) => {
  return (
    <View key={index} style={styles.container}>
      <ImageBackground
        source={require('../images/none.png')}
        imageStyle={{borderRadius: 15}}
        style={styles.background}>
        <Image style={styles.image} source={{uri: item.image}} />
        <View style={styles.textView}>
          <Text style={styles.text}>{trim(item.title, 50)}</Text>
          <Text style={styles.description}>{trim(item.description, 85)}</Text>
          <TouchableOpacity
            style={styles.link}
            onPress={() => Linking.openURL(item.url)}>
            <Text style={styles.linkText}>Read On</Text>
          </TouchableOpacity>
        </View>
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '95%',
    marginTop: '5%',
    marginBottom: '5%',
    alignSelf: 'center',
    flexDirection: 'column',
    borderRadius: 15,
  },
  background: {
    flex: 1,
    justifyContent: 'center',
    height: '100%',
    width: '100%',
  },
  image: {
    borderRadius: 15,
    height: '100%',
    width: '100%',
    alignSelf: 'center',
  },
  textView: {
    flex: 1,
    height: '45%',
    marginTop: '-60%',
    backgroundColor: 'rgba(187,225,250,0.88)',
    borderBottomRightRadius: 15,
    borderBottomLeftRadius: 15,
  },
  text: {
    color: '#0f4c75',
    fontSize: 25,
    fontWeight: 'bold',
    textAlign: 'center',
    marginHorizontal: '3%',
    marginVertical: '5%',
  },
  description: {
    color: '#0f4c75',
    fontSize: 18,
    fontStyle: 'italic',
    textAlign: 'center',
    marginHorizontal: '1%',
  },
  link: {
    opacity: 1.5,
    position: 'absolute',
    bottom: '5%',
    width: '95%',
    borderRadius: 10,
    backgroundColor: '#0f4c75',
    marginHorizontal: '1%',
    alignSelf: 'center',
  },
  linkText: {
    color: '#bbe1fa',
    padding: '3%',
    textAlign: 'center',
    fontSize: 23,
    fontWeight: 'bold',
  },
});

export default CarouselItem;
