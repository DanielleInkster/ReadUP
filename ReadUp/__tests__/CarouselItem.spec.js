import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';
import CarouselItem from '../src/components/CarouselItem';

import renderer from 'react-test-renderer';

const article = {
  id: '1',
  title: 'hello',
  description: 'world',
  url: 'www.fakesite.com',
  image: '../src/images/none.jpg',
};
let mockOpenURL = jest.fn();

jest.mock('react-native/Libraries/Linking/Linking', () => ({
  openURL: mockOpenURL,
}));

test('renders correctly', () => {
  const tree = renderer
    .create(<CarouselItem item={article} index={1} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('CarouselItem', () => {
  it('displays article info', () => {
    const {getByText} = render(<CarouselItem item={article} index={1} />);
    expect(getByText('hello')).not.toBeNull();
    expect(getByText('world')).not.toBeNull();
  });
});

describe('Read On button', () => {
  it('opens Link onClick', () => {
    const {getByTestId} = render(<CarouselItem item={article} index={1} />);
    fireEvent.press(getByTestId('button'));
    expect(mockOpenURL).toHaveBeenCalled();
  });
});
