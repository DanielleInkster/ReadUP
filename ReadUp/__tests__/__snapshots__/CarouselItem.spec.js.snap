// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "borderRadius": 15,
      "flex": 1,
      "flexDirection": "column",
      "marginBottom": "5%",
      "marginTop": "5%",
      "width": "95%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/none.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          Object {
            "borderRadius": 15,
          },
        ]
      }
    />
    <Image
      source={
        Object {
          "uri": "../src/images/none.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 15,
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(187,225,250,0.88)",
          "borderBottomLeftRadius": 15,
          "borderBottomRightRadius": 15,
          "flex": 1,
          "height": "45%",
          "marginTop": "-52%",
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#0f4c75",
            "fontFamily": "sans-serif-medium",
            "fontSize": 25,
            "fontWeight": "bold",
            "marginBottom": "2%",
            "marginHorizontal": "3%",
            "marginTop": "1%",
            "textAlign": "center",
          }
        }
      >
        hello
      </Text>
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#0f4c75",
            "fontFamily": "sans-serif-medium",
            "fontSize": 18,
            "fontStyle": "italic",
            "marginHorizontal": "3%",
            "textAlign": "center",
          }
        }
      >
        world
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#0f4c75",
            "borderRadius": 10,
            "bottom": "5%",
            "marginHorizontal": "1%",
            "opacity": 1.5,
            "position": "absolute",
            "width": "95%",
          }
        }
        testID="button"
      >
        <Text
          style={
            Object {
              "color": "#bbe1fa",
              "fontFamily": "sans-serif-medium",
              "fontSize": 23,
              "fontWeight": "bold",
              "padding": "3%",
              "textAlign": "center",
            }
          }
        >
          Read On
        </Text>
      </View>
    </View>
  </View>
</View>
`;
